docs/bold-bi-auto-deployment.md

```diff
@@ -0,0 +1,983 @@
+
+## Deployment prerequisites
+
+The following requirements are necessary to deploy the Bold BI solution using kubectl. 
+
+* [Install Kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) to deploy Bold BI using kubectl.
+* [File Storage](pre-requisites.md#file-storage)
+* [Create and connect a cluster](pre-requisites.md#create-a-cluster)
+* Load Balancing- [Nginx](https://kubernetes.github.io/ingress-nginx/deploy/)
+
+## Steps for Bold BI auto deployment using kubectl.
+
+The following links explain Bold BI Kubernetes deployment in a specific cloud environments.
+
+
+ * [Azure Kubernetes Service(AKS)](bold-bi-auto-deployment.md#bold-bi-on-microsoft-azure-kubernetes-service)
+ * [Amazon Elastic Kubernetes Service(EKS)](bold-bi-auto-deployment.md#bold-bi-on-amazon-elastic-kubernetes-service)
+ * [Google Kubernetes Engine (GKE)](bold-bi-auto-deployment.md#bold-bi-on-google-kubernetes-engine)
+
+
+### Bold BI on Microsoft Azure Kubernetes Service
+
+1. Download the deployment file from [here](https://raw.githubusercontent.com/boldbi/boldbi-kubernetes/UMP-22952-Autodeployment-documentation/deploy/auto-deployment/deploy_aks.yaml) to deloy Bold BI on AKS.
+
+2. Navigate to the folder where the deployment files were downloaded from **Step 1**.
+
+3. Create a File share instance in your storage account and note the File share name to store the shared folders for application usage.
+
+4. Encode the storage account name and storage key in base64 format.
+  
+   For encoding the values to base64 please run the following command in powershell
+
+    ```console
+    [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes("<plain-text>"))
+    ```
+
+   ![File Share details](images/deploy/aks/aks-file-storage.png)
+
+5. Open **deploy_aks.yaml** file, downloaded in **Step 1**. Replace the **base64 encoded storage account name**, **base64 encoded storage account key**, and **File share name** which are noted in above steps to `<base64_azurestorageaccountname>`, `<base64_azurestorageaccountkey>`, and `<file_share_name>` places in the file respectively. You can also change the storage size in the YAML file.
+
+    ![PV Claim](images/deploy/aks/pv_claim.png)
+
+6. After connecting with your cluster, deploy the latest Nginx ingress controller to your cluster using the following command.
+
+   ```sh
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.2.0/deploy/static/provider/cloud/deploy.yaml
+   ```
+
+7. Enter the variable information needed for complete the auto deployment in <b>deploy_aks.yaml</b> file as shown below.
+
+    * Enter the Bold BI licence key, user and database server details.
+        
+        ![Licence-and-User-Details](images/deploy/aks/licence-and-user-details.png)
+
+        ## Environment variables details for configuring `Application Startup` in backend
```

Comment:
Environment variables details for configuring `Application Startup` in the backend.

Reply:
concern addressed